lib/google_directions_api/directions.rb

```diff
@@ -55,43 +55,31 @@ def data
     end
 
     def total_distance
-      if waypoints_present?
-        meters = 0
-        data["routes"][0]["legs"].each do |leg|
-          meters += leg["distance"]["value"]
-        end
-        return meters
-      else
-        data["routes"][0]["legs"][0]["distance"]["value"]
+      meters = 0
+      data["routes"][0]["legs"].each do |leg|
+        meters += leg["distance"]["value"]
       end
+
+      meters
     end
 
     def total_duration
-      if waypoints_present?
-        seconds = 0
-        data["routes"][0]["legs"].each do |leg|
-          seconds += leg["duration"]["value"]
-        end
-        return seconds
-      else
-        data["routes"][0]["legs"][0]["duration"]["value"]
+      seconds = 0
+      data["routes"][0]["legs"].each do |leg|
+        seconds += leg["duration"]["value"]
       end
+
+      seconds
     end
 
     def tolls_along_route?
-      if waypoints_present?
-        data["routes"][0]["legs"].each do |leg|
-          next unless leg["steps"].any?
-
-          if leg["steps"]["html_instructions"].try(:downcase).try(:include?, 'toll road')
```

Comment:
Note to self: The original problem was here - `html_instructions` sub-element is an array (see line 93 for correct usage)